File: backend/src/main/java/turip/place/controller/dto/response/PlaceResponse.java
package turip.place.controller.dto.response;

import java.util.List;
import turip.place.domain.Place;
import turip.placecategory.domain.PlaceCategory;

public record PlaceResponse(
        Long id,
        String name,
        String url,
        String address,
        double latitude,
        double longitude,
        List<CategoryResponse> categories
) {
    public static PlaceResponse from(Place place) {
        return new PlaceResponse(
                place.getId(),
                place.getName(),
                place.getUrl(),
                place.getAddress(),
                place.getLatitude(),
                place.getLongitude(),
                CategoryResponse.from(
                        place.getPlaceCategories().stream()
                                .map(PlaceCategory::getCategory)
                                .toList()
                )
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/content/place/dto/PlaceResponse.kt
package com.on.turip.data.content.place.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PlaceResponse(
    @SerialName("address")
    val address: String,
    @SerialName("categories")
    val categories: List<CategoryResponse>,
    @SerialName("id")
    val id: Long,
    @SerialName("latitude")
    val latitude: Double,
    @SerialName("longitude")
    val longitude: Double,
    @SerialName("name")
    val name: String,
    @SerialName("url")
    val url: String,
)
